crates/bevy_ui/src/layout/ui_surface.rs

```diff
@@ -115,18 +152,19 @@ impl UiSurface {
 
     /// Update the children of the taffy node corresponding to the given [`Entity`].
     pub fn update_children(&mut self, entity: Entity, children: impl Iterator<Item = Entity>) {
-        self.taffy_children_scratch.clear();
-
-        for child in children {
-            if let Some(taffy_node) = self.entity_to_taffy.get(&child) {
-                self.taffy_children_scratch.push(*taffy_node);
-            }
-        }
+        let children = children
+            .map(|child| {
+                self.entity_to_taffy
+                    .get(&child)
+                    .cloned()
+                    .unwrap_or_else(|| {
+                        panic!("failed to resolve taffy id for child entity {child} in {entity}")
+                    })
```

Comment:
Is it possible to rewrite this using `Option::expect()`?
```suggestion
                    .expect("failed to resolve taffy id for child entity {child} in {entity}")
```